src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -589,57 +589,57 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getFilterLogs(UInt256 filterId)
 
     public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
     {
+        BlockParameter fromBlock = filter.FromBlock;
+        BlockParameter toBlock = filter.ToBlock;
+
         // because of lazy evaluation of enumerable, we need to do the validation here first
         CancellationTokenSource timeout = BuildTimeoutCancellationTokenSource();
         CancellationToken cancellationToken = timeout.Token;
 
+        if (!TryFindOrUseLatest(_blockFinder, ref toBlock, out SearchResult<BlockHeader> toBlockResult, out long sourceToBlockNumber))
+        {
+            timeout.Dispose();
```

Comment:
What do you think about adding a `finally` with `timeout.Dispose()` and remove all other occurrences?

Reply:
Good idea